### PR TITLE
Feature/173964655 add proteomics differential

### PR DESCRIPTION
**Pull request by lingyun1010**
This PR goes with web-core PR https://github.com/ebi-gene-expression-group/atlas-web-core/pull/69.

### CODE REVIEW

File: app/src/main/java/uk/ac/ebi/atlas/trader/GxaExperimentRepository.java [hunk 1]
```diff
@@ -89,6 +89,12 @@ public Experiment getExperiment(String experimentAccession) {
                         experimentDesign,
                         idfParserOutput,
                         ImmutableSet.of("Microarray 2-colour mRNA"));
+            case PROTEOMICS_DIFFERENTIAL:
+                return rnaSeqDifferentialExperimentFactory.create(
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/trader/GxaExperimentRepository.java]: Inject and use `proteomicsDifferentialExperimentFactory`.

File: app/src/test/java/uk/ac/ebi/atlas/trader/GxaExperimentRepositoryIT.java [hunk 2]
```diff
@@ -20,12 +20,7 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.MICROARRAY_1COLOUR_MICRORNA_DIFFERENTIAL;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.MICROARRAY_1COLOUR_MRNA_DIFFERENTIAL;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.MICROARRAY_2COLOUR_MRNA_DIFFERENTIAL;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.PROTEOMICS_BASELINE;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.RNASEQ_MRNA_BASELINE;
-import static uk.ac.ebi.atlas.model.experiment.ExperimentType.RNASEQ_MRNA_DIFFERENTIAL;
+import static uk.ac.ebi.atlas.model.experiment.ExperimentType.*;
```
Comment (alfonsomunozpomer) [inline app/src/test/java/uk/ac/ebi/atlas/trader/GxaExperimentRepositoryIT.java:23]: Don’t use wildcard imports (fix your settings in IDEA if this was your IDE “helping” you out).

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/JsonDifferentialExperimentController.java [hunk 3]
```diff
@@ -125,4 +125,17 @@ public String differentialRnaSeqExperimentData(
                 (DifferentialExperiment) experimentTrader.getExperiment(experimentAccession, accessKey),
                 accessKey, preferences));
     }
+
+    @RequestMapping(value = "/json/experiments/{experimentAccession}",
+            produces = "application/json;charset=UTF-8",
+            params = "type=PROTEOMICS_DIFFERENTIAL")
+    @ResponseBody
+    public String differentialProteomicsExperimentData(
+            @ModelAttribute("preferences") @Valid MicroarrayRequestPreferences preferences,
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/JsonDifferentialExperimentController.java]: Why `MicroArrayRequestPreferences`? Please use a suitable class, create it if necessary.

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/JsonDifferentialExperimentController.java [hunk 4]
```diff
@@ -125,4 +125,17 @@ public String differentialRnaSeqExperimentData(
...
+            @PathVariable String experimentAccession,
+            @RequestParam(defaultValue = "") String accessKey) {
+        return GSON.toJson(diffRnaSeqExperimentPageService.getResultsForExperiment(
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/JsonDifferentialExperimentController.java]: If it’s a proteomics experiment the name should reflect it.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
As with the companion PR, please be more observant of the relationship between the experiment type and the class/variable names that deal with them.

**Comment by upendrakumbham** [comment]
I just tested the application with your latest core commit. Now experiments page is fine, I could see a list of experiments along with `E-PROT-39` experiment as well with Proteomics differential technology type.

I could see the result page as well after clicking `E-PROT-39` experiment

image

**Comment by upendrakumbham** [comment]
The above 4 buttons are not working for me. Please check from your side.

**Review by upendrakumbham — APPROVED** [review]
LGTM! I tested PR on my local machine. All issues got fixed.

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadController.java [hunk 5]
```diff
@@ -66,16 +66,16 @@ public ExperimentDownloadController(ExperimentTrader experimentTrader,
                                                 proteomicsExperimentDownloadSupplier,
                                         ExperimentDownloadSupplier.RnaSeqBaseline
                                                 rnaSeqBaselineExperimentDownloadSupplier,
-                                        ExperimentDownloadSupplier.RnaSeqDifferential
-                                                rnaSeqDifferentialExperimentDownloadSupplier,
+                                        ExperimentDownloadSupplier.BulkDifferential
+                                                    bulkDifferential,
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentDownloadController.java]: This variable should be named `bulkDifferentialExperimentDownloadSupplier` to be consistent with the others.

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java [hunk 6]
```diff
@@ -55,7 +55,7 @@
     public ExpressionAtlasContentService(
             ExperimentDownloadSupplier.Proteomics proteomicsExperimentDownloadSupplier,
             ExperimentDownloadSupplier.RnaSeqBaseline rnaSeqBaselineExperimentDownloadSupplier,
-            ExperimentDownloadSupplier.RnaSeqDifferential rnaSeqDifferentialExperimentDownloadSupplier,
+            ExperimentDownloadSupplier.BulkDifferential bulkDifferential,
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java:58]: Same as before, please rename to `bulkDifferentialExperimentDownloadSupplier`.

File: app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/opentargets/OpenTargetsEvidenceController.java [hunk 7]
```diff
@@ -38,22 +38,22 @@
 @Scope("request")
 public class OpenTargetsEvidenceController extends JsonExperimentController {
     private final
-    EvidenceService<DifferentialExpression, DifferentialExperiment, RnaSeqRequestContext, RnaSeqProfile>
+    EvidenceService<DifferentialExpression, DifferentialExperiment, BulkDifferentialRequestContext, BulkDifferentialProfile>
```
Comment (alfonsomunozpomer) [inline app/src/main/java/uk/ac/ebi/atlas/experimentpage/json/opentargets/OpenTargetsEvidenceController.java:41]: Rename `diffRnaSeqEvidenceService`  to reflect the change in type. There are also other variables throughout this class with the same issue, please go through it carefully and change all references to `diffRnaSeq` elements.

File: app/src/test/java/uk/ac/ebi/atlas/experimentpage/FileDownloadControllerTest.java [hunk 8]
```diff
@@ -47,12 +47,11 @@
     private ExperimentDownloadSupplier.RnaSeqBaseline rnaSeqBaselineExperimentDownloadSupplier;
 
     @Mock
-    private ExperimentDownloadSupplier.RnaSeqDifferential rnaSeqDifferentialExperimentDownloadSupplier;
+    private ExperimentDownloadSupplier.BulkDifferential bulkDifferential;
```
Comment (alfonsomunozpomer) [inline app/src/test/java/uk/ac/ebi/atlas/experimentpage/FileDownloadControllerTest.java]: Again, this should be `bulkDifferentialExperimentDownloadSupplier`.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
Just some renaming business left to do. My only concern logic-wise would be to check if `ExperimentDownloadSupplier.Proteomics`  is a baseline-only related class. If that’s the case we should rename it so we don’t get confused about whether to use `ExperimentDownloadSupplier.BulkDifferential` when we have a differential proteomics experiment. If this is unclear please let me know.

**Review by alfonsomunozpomer — APPROVED** [review]
All fine. Remember to add the ticket we discussed to rename the proteomics baseline classes that are only denoted with `Proteomics`. Good job!

**Pull request merged**